Code/Tools/FBuild/FBuildCore/Graph/ObjectNode.cpp

```diff
@@ -1748,10 +1748,13 @@ bool ObjectNode::BuildArgs( const Job * job, Args & fullArgs, Pass pass, bool us
                 // we expand relative includes as they would be on the host (so the remote machine's
                 // working dir is not used, which might be longer, causing this overflow an internal
                 // limit of cl.exe)
-                if ( ( job->IsLocal() == false ) && IsStartOfCompilerArg_MSVC( token, "I" ) )
+                if ( ( job->IsLocal() == false ) && (
+                        IsStartOfCompilerArg_MSVC( token, "I" ) ||
+                        IsStartOfCompilerArg_MSVC( token, "imsvc" ) ||
+                        IsStartOfCompilerArg_MSVC( token, "external:I" ) ) )
                 {
                     // Get include path part
-                    const char * start = token.Get() + 2; // Skip /I or -I
+                    const char * start = token.Get() + 2; // Skip /I or -I or /imsvc or -imsvc or /external:I or -external:I
```

Comment:
This still seems to add +2 to the token.Get(), which doesn't seem correct for "imsvc" and "external:I"

Reply:
Yes you are absolutely right,
I'll update this patch soon after additional testing.